### ISSUE TITLE
rule proposal: disallow wider return type annotations

### ISSUE THREAD

**Issue by gameroman** [oxc-project/tsgolint#694]
### Invalid

```ts
// explicit return type is wider than what the function actually returns

function a(): string | number {
  return "some string";
}

declare const v: 1 | 2;

function b(): 1 | 2 | 3 {
  return v;
}
```

### Valid

```ts
// explicit return type matches actual return

function a(): string {
  return "some string";
}

declare const v: 1 | 2;

function b(): 1 | 2 {
  return v;
}
```

**Comment by mchevestrier** [comment]
FWIW, I added such a rule in my own ESLint plugin: [`wyrm/no-lax-return-type`](https://github.com/mchevestrier/eslint-plugin-wyrm/blob/master/packages/eslint-plugin-wyrm/docs/rules/no-lax-return-type.md)

See also [`wyrm/no-lax-array-type`](https://github.com/mchevestrier/eslint-plugin-wyrm/blob/master/packages/eslint-plugin-wyrm/docs/rules/no-lax-array-type.md)
